documentation/projects/proposals/trust_and_safety/content_report_moderation/20231208-implementation_plan_django_admin_moderator_access.md

```diff
@@ -0,0 +1,587 @@
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+We'll implement this plan in the following order. I've noted which items block
+others or are non-blocking.
+
+1. Preliminary access control (somewhat blocking `ModerationDecision`)
+2. Add user preferences admin view (non-blocking)
+3. Report table view improvements (non-blocking)
+4. `ModerationDecision` (blocking media moderation view)
+5. Media moderation view
+6. The view itself
+7. The actions
+8. The moderation soft-lock
+9. Finalise access control
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+### 1. Preliminary access control
+
+[See "Access control" for rationale](#access-control)
+
+- Add `User` and `Group` admin views
+- Create the "Content Moderator" group and add basic preliminary permissions to
+  read and update content reports and media items
+  - We'll expand these later on in the implementation as needed
+
+### 2. Add user preferences admin view
+
+[See "Moderator preferences" for rationale](#moderator-preferences)
+
+- Sub-class the `User` model to create `OpenverseUser` and add a new binary JSON
+  column, `preferences`
+- Create a new admin view for `OpenverseUser` called `User preferences` that has
+  only a single checkbox control to turn on and off the `moderator.blur_images`
+  setting
+
+### 3. Table view improvements
+
+[See "Table view improvements" for rationale](#table-view-improvements)
+
+- Create a new view based on the media object default model view but sort media
+  by pending report count and the oldest report's age
+- Exclude media items that have no reports
+- Filter to only show media with pending reports by default, but allow showing
+  all reported media
+
+### 4. `ModerationDecision`
+
+[See "`ModerationDecision`" for rationale](#moderationdecision)
+
+- Create the new `ModerationDecision` models
+  - Create an `AbstractModerationDecision` for each media type
+- Add new `decision_id` column to report models
+- Remove bulk report updates from the report models' `save` methods
+
+### 5. Media moderation view
+
+[See "Media moderation view" for rationale](#media-moderation-view)
+
+This is in three separate issues:
+
+1. Create a new media view to replace the existing view
+
+- Use the basic model admin but with only "view" settings for all model fields
+- We'll need to expand the rendered fields to include things like the rendered
+  (interactive) thumbnail and audio player
```

Comment:
"interactive" thumbnail? Does that mean that thumbnail can be blurred as well?

Reply:
Yes. That's covered in this implementation plan (see the moderator preferences section, for example). The project plan calls for this.